pkg/eestream/aesgcm.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package eestream
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+)
+
+type aesgcmEncrypter struct {
+	blockSize     int
+	key           [32]byte
+	startingNonce [12]byte
+	overhead      int
+	//aesgcm        AEAD
+}
+
+// NewAesGcmEncrypter returns a Transformer that encrypts the data passing
+// through with key.
+//
+// startingNonce is treated as a big-endian encoded unsigned
+// integer, and as blocks pass through, their block number and the starting
+// nonce is added together to come up with that block's nonce. Encrypting
+// different data with the same key and the same nonce is a huge security
+// issue. It's safe to always encode new data with a random key and random
+// startingNonce. The monotonically-increasing nonce (that rolls over) is to
+// protect against data reordering.
+//
+// When in doubt, generate a new key from crypto/rand and a startingNonce
+// from crypto/rand as often as possible.
+func NewAesGcmEncrypter(key *[32]byte, startingNonce *[12]byte,
+	encryptedBlockSize int) (Transformer, error) {
+	ptrkey := *key
+	slicekey := ptrkey[:]
+	block, err := aes.NewCipher(slicekey)
```

Comment:
You can just pass `(*key)[:]` here instead of introducing two new local variables.

Reply:
modified as suggested